### ISSUE TITLE
Kraken fetch_balance missing coin mapping

### ISSUE THREAD

**Issue by xmatthias** [ccxt/ccxt#24663]
### Operating System

linux

### Programming Languages

Python

### CCXT Version

4.4.40

### Description

when kraken ["rewards"](https://www.kraken.com/features/rewards) are aktivated, balances for supported currencies will report under "xxx.F" - intead of under their real symbol.

So far, that's no problem - and from the response - we can see that there's both ATOM and ATOM.F listed - with the full balance being on ATOM.F (this is expected, rewards is always "all in" as far as i could tell).

Now the problem is with currencies with a non-default symbol on kraken (e.g. XBT) - which ccxt correctly maps to BTC - but is missing the mapping for XBT.F (should map to BTC.F) - see response XBT.F vs. BTC.

I'm not sure if the fix is as simple as adding the ".F" variants to `commonCurrencies` (https://github.com/ccxt/ccxt/blob/c4c744366dc056f951839058974a38a40d430ad1/ts/src/kraken.ts#L230C1-L237C31) - or if that's the wrong approach and would have unintended side-effects.

### Code

fetch_balance response

``` python
import ccxt
exchange = ccxt.kraken({
    'apiKey': '<yourApiKey>', 
    'secret': '<yoursecret>', 
    })
res = exchange.fetch_balances()
print(res)
``` 

(a slightly reduced response sample below).

```
{'timestamp': None, 'datetime': None, 
     'ATOM': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'ATOM.F': {'free': 0.00347846, 'used': 0.0, 'total': 0.00347846
    }, 'TRX': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'TRX.F': {'free': 39.5972171, 'used': 0.0, 'total': 39.5972171
    }, 'USDT': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'USDT.F': {'free': 6.016e-05, 'used': 0.0, 'total': 6.016e-05
    }, 'XBT.F': {'free': 0.0160225084, 'used': 0.0, 'total': 0.0160225084
    }, 'XTZ': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'XTZ.F': {'free': 6.98581419, 'used': 0.0, 'total': 6.98581419
    }, 'BTC': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'DOGE': {'free': 0.00656161, 'used': 0.0, 'total': 0.00656161
    }, 'XLM': {'free': 9e-08, 'used': 0.0, 'total': 9e-08
    }, 'XMR': {'free': 0.06627103, 'used': 0.0, 'total': 0.06627103
    }, 'XRP': {'free': 0.0, 'used': 0.0, 'total': 0.0
    }, 'ZEC': {'free': 3e-08, 'used': 0.0, 'total': 3e-08
    }, 'EUR': {'free': 423.9351, 'used': 0.0, 'total': 423.9351
    }
}
2

```

**Comment by ttodua** [comment]
@xmatthias thanks for report, let me check and give you an answer shortly

**Comment by ttodua** [comment]
very concerning that US exchanges also have such messy and frustrating APIs, either one or another workaround for this problem has its own disadvantages and I am still trying to find out which would be best way to fix the problem

**Comment by xmatthias** [comment]
yeah i've been scared to simply add xbt.f -> btc.f as mapping to commonCurrencies exactly because of this :laughing: 

If it's causing problems - i can also work around this at the freqtrade level - though i thought it'd be beneficial to have this fixed / corrected upstream ;)

**Comment by ttodua** [comment]
I've attached a fix, because for me, it seems specific to "balance" endpoint, so I think this should have a local workaround in parseBalance, so we avoid touching `commonCurrencies` at this moment.
if you find out that the currencies are also affected in other areas (after that "earnings activation") from other endpoints too,  let me know

**Comment by xmatthias** [comment]
so far, i didn't see ".F" currencies anywhere else fortunately :laughing: so the pairs, orders, etc. still work as expected.

**Comment by Montiwa11** [comment]
}, 'XTZ.F': {'free': 6.98581419, 'used': 0.0, 'total': 6.98581419
[res]()
